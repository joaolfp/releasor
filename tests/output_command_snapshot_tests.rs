use std::fs;
use insta::assert_snapshot;
use releasor::OutputCommand;

#[test]
fn test_cargo_release_output_snapshot() {
    let output = OutputCommand::cargo_release_output();
    let stdout = String::from_utf8_lossy(&output.stdout);
    let stderr = String::from_utf8_lossy(&output.stderr);

    let snapshot = format!(
        "status: {}\nstdout:\n{}\nstderr:\n{}",
        output.status, stdout, stderr
    );

    assert_snapshot!("cargo_build_output", snapshot);
}

#[test]
fn test_tar_output_snapshot() {
    let output = OutputCommand::tar_output(
        "releasor.tar.gz",
        "releasor"
    );

    let stdout = String::from_utf8_lossy(&output.stdout);
    let stderr = String::from_utf8_lossy(&output.stderr);

    let snapshot = format!(
        "status: {}\nstdout:\n{}\nstderr:\n{}",
        output.status, stdout, stderr
    );

    assert_snapshot!("tar_output", snapshot);

    fs::remove_file("releasor.tar.gz")
        .expect("Failed to delete the file");
}