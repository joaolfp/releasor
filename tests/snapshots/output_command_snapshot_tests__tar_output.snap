---
source: tests/output_command_snapshot_tests.rs
expression: snapshot
snapshot_kind: text
---
status: exit status: 0
stdout:

stderr:
a releasor
